### DISCUSSION TITLE
how to run evaluation.py by myself?

### DISCUSSION THREAD

**Discussion by greatofdream** [PaddlePaddle/PaddleSpatial#167] [BaiduKDDCup2022] [answered]
The `evalution.py` in demo doesn't contain any main program, should I configure one function to run  `evalution.py`？

https://github.com/PaddlePaddle/PaddleSpatial/blob/main/apps/wpf_baseline_gru/kddcup22-sdwpf-evaluation/paddlepaddle/evaluation.py

**Comment by ramber1836** [comment]
You can add the initial statements in the evaluation.py to invoke the eval() interface and run this script by yourself.

**Comment by shaido987** [comment, marked as answer]
You can add the following to the evaluation.py file:

```python
if __name__ == '__main__':
    evaluate("./initial-test-run"
```

Then put the prepare.py, predict file, etc. in a subfolder called initial-test-run (or whatever you want to call it). Now you can run the evaluation.py file to make sure that your submission code works.

It is also possible to use the `eval` function but then you need to create a zip file with the code first.
